db/wal_manager.cc

```diff
@@ -173,7 +173,6 @@ void WalManager::PurgeObsoleteWALFiles() {
   if (!s.ok()) {
     ROCKS_LOG_ERROR(db_options_.info_log, "Can't get archive files: %s",
                     s.ToString().c_str());
-    assert(false);
```

Comment:
Bug fix where this path can happen in dbg now with error injection in stress test.